.github/workflows/build.yml

```
@@ -41,53 +41,36 @@ on:
       - '.github/workflows/build.yml'
 
 jobs:
-  build-linux:
+  build:
     strategy:
       fail-fast: false
       matrix:
-        shared: [0, 1]
-    name: Build Linux (${{ matrix.shared == 0 && 'Static' || 'Shared' }})
-    runs-on: ubuntu-latest
+        include:
+          - os: ubuntu-latest
+            os_name: linux
+            shared: 0
+          - os: ubuntu-latest
+            os_name: linux
+            shared: 1
+          - os: macos-latest
+            os_name: macos
+            shared: 0
+          - os: macos-latest
+            os_name: macos
+            shared: 1
+          - os: windows-latest
+            os_name: windows
+            shared: 0
+          - os: windows-latest
+            os_name: windows
+            shared: 1
```

```suggestion
        shared: [0, 1]
        include:
          - os: ubuntu-latest
            os_name: Linux
            os_name_lowercase: linux
          - os: macos-latest
            os_name: MacOS
            os_name_lowercase: macos
          - os: windows-latest
            os_name: Windows
            os_name_lowercase: windows
```

Cannot we do it like that? See [Expanding or adding matrix configurations](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#expanding-or-adding-matrix-configurations) and [Example: Expanding configurations](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#example-expanding-configurations).